plugins/cindy3d/src/js/PrimitiveRenderer.js

```diff
@@ -219,9 +229,22 @@ PrimitiveRenderer.prototype.addPrimitive = function(attributes) {
     throw new GlError("Wrong number of attributes given");
   if (this.count == this.capacity) {
     let c = this.capacity*2, nd, nda, ndi, i, j, k, o, e = this.elements;
+    if (c > this.maxCount) {
+      if (this.capacity < this.maxCount) {
+        c = this.maxCount;
+      } else {
+        this.indexByteCount = 4;
+        this.itemTotalByteCount = this.itemAttribByteCount +
+          this.numElements * this.indexByteCount;
+        this.maxCount = Math.floor((1 << 31) / this.itemAttribByteCount);
```

Comment:
Actually, we would like to have `2^32` here. With the division we will use doubles anyway.
So we could simply use `Math.pow(2,32)`, or if you like bit-shifts `2*((-1^(1<<31))+1)` ;-)